doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,243 @@
+***************************
+
+Ubuntu installation automation is performed with the autoinstall format.
+You might also know this feature as unattended or handsoff or preseeded
+installation.
+
+This format is supported in the following installers:
+ * Ubuntu Server, version 20.04 and later
+ * Ubuntu Desktop, version 23.04 and later
+
+Autoinstallation lets you answer all those configuration questions ahead of
+time with an *autoinstall config* and lets the installation process run without
+any interaction.
+
+
+Differences from debian-installer preseeding
+============================================
+
+*preseeds* are the way to automate an installer based on debian-installer
+(also known as d-i).
+
+Autoinstalls differ from preseeds in the following ways:
+ * the format is different (yaml vs debconf-set-selections)
+ * when the answer to a question is not present in a preseed, d-i stops and
+   asks the user for input. Autoinstalls are not like this:  by default, if
+   there is any autoinstall configuration at all, the installer takes the
+   default for any unanswered question (and fails if there is no default).
+ * You can designate particular sections in the configuration as "interactive",
+   which means the installer will still stop and ask about those.
+
+
+Providing the autoinstall configuration
+=======================================
+
+There are 2 methods of providing the autoinstall configuration:
+ * Carried as part of cloud-config
+ * Directly on the install media
+
+Autoinstall by way of cloud-config
+----------------------------------
+
+The suggested method of providing autoinstall to the Ubuntu installer is by way
+of cloud-init.  This allows the configuration to be applied to the installer
+without having to modify the install media.
+
+The autoinstall config is provided via cloud-init configuration, which is
+almost endlessly flexible. In most scenarios the easiest way will be to provide
+user-data via the :external+cloud-init:ref:`datasource_nocloud` data source.
+
+When providing autoinstall via cloud-init, the autoinstall config is provided
+as :external+cloud-init:ref:`user_data_formats-cloud_config`.  This
+means we need a :code:`#cloud-config` header. The autoinstall directives are
+placed under a top level :code:`autoinstall` key, like so:
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+        version: 1
+        ....
+
+Autoinstall on the install media
+--------------------------------
+
+Another option for supplying autoinstall to the Ubuntu installer is to place a
+file named :code:`autoinstall.yaml` on the install media itself.
+
+There are two potential locations for the :code:`autoinstall.yaml` file:
+ * At the root of the "cdrom".  When writing the installation ISO to a USB
+   Flash Drive, this can be done by copying the :code:`autoinstall.yaml` to the
+   partition containing the contents of the ISO - i.e., in the same directory
+   containing the :code:`casper` directory.
```

Comment:
"In the same directory containing the :code:`casper` directory" was a bit confusing to me. I'd suggest something like "In the directory containing the :code:`casper` sub-directory".

Reply:
Done